File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program24.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept temperature in Fahrenheit and convert it into celsius. 
//		(1 celsius = (Fahrenheit -32) * (5/9))
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Demo
{
	private:
		float fTemp;
		double dConvert;
	
	public:
		Demo(float fTemp,double dConvert=0.0)
		{
			this->fTemp = fTemp;
		}

		double FhtoCs(float fTemp)
		{
			dConvert = (fTemp-32)*5/9;
			return dConvert;
		}	
};

int main()
{
	float fValue = 0.0;
	double dRet = 0.0;

	cout<<"Enter temperature in Fahrenheit: ";
	cin>>fValue;

	Demo *dObj = new Demo(fValue);
	dRet = (*dObj).FhtoCs(fValue);
	cout<<dRet<<"\n";

	return 0;
}

/*

	OUTPUT:

		Enter temperature in Fahrenheit: 10
		-12.222222

		Enter temperature in Fahrenheit: 34
		1.111111

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program24.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept temperature in Fahrenheit and convert it into celsius. 
//		(1 celsius = (Fahrenheit -32) * (5/9))
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public float iValue;

	public Solution(float iValue)
	{
		this.iValue = iValue;
	}

	public double FhToCs()
	{

                double dValue =  (iValue-32)*5/9;      	
		return dValue;
	
	}
}

class Program4
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		double dRet=0.0;

		System.out.println("Enter temperature in Fahrenheit: ");
		float fValue = sObj.nextFloat();

		Solution obj = new Solution(fValue);
		dRet = obj.FhToCs();
	
		System.out.println("Conversion is: "+dRet);
	}
}

/*
OUTPUT:

Enter temperature in Fahrenheit: 
10
Conversion is: -12.222222328186035

*/
